### ISSUE TITLE
MS SQL varchar(max) not mapped to propper PostgreSQL data type

### ISSUE THREAD

**Issue by ninsbl** [evolute-pt/dbtransfer#13]
Hi,

I am trying to transfer a MS SQL database to PostgreSQL, which seems to work fine, until the process reaches a table which contains a column of data type "varchar(max)". This data type gets mapped to varchar( 2147483647), which then causes an error in CREATE TABLE:

"org.postgresql.util.PSQLException: ERROR: length for type varchar cannot exceed 10485760"

And the process stops.

Maybe better to map "vachar(max)" to "text"?

**Comment by luiscamposflores** [comment]
Hello,

Yes, any VARCHAR with length above 10485760 should be mapped to TEXT, I will implement that fix soon.

**Comment by CheeseL0ver** [comment]
I just ran into this issue. Is there any ideas as to when a fix can be implemented? @luiscamposflores

**Comment by smarques** [comment]
@CheeseL0ver did you find a way to fix this? I am stuck with the same problem

**Comment by CheeseL0ver** [comment]
@smarques No, the fix @luiscamposflores planned to implement does not seem to have been done. We ended up just rebuilding the database instead of using the conversion tool.
